packages/labs/signals/src/lib/signal-watcher.ts

```diff
@@ -4,23 +4,79 @@
  * SPDX-License-Identifier: BSD-3-Clause
  */
 
-import type {PropertyDeclaration, PropertyValueMap, ReactiveElement} from 'lit';
+import type {ReactiveElement} from 'lit';
 import {Signal} from 'signal-polyfill';
-import {WatchDirective} from './watch.js';
-
-type ReactiveElementConstructor = abstract new (
-  // eslint-disable-next-line @typescript-eslint/no-explicit-any
-  ...args: any[]
-) => ReactiveElement;
 
 export interface SignalWatcher extends ReactiveElement {
-  _updateWatchDirective(d: WatchDirective<unknown>): void;
-  _clearWatchDirective(d: WatchDirective<unknown>): void;
+  _watcher?: Signal.subtle.Watcher;
+}
+
+interface EffectOptions {
+  beforeUpdate?: boolean;
+  manualDispose?: boolean;
+}
+
+export interface ElementEffectOptions extends EffectOptions {
+  element?: SignalWatcher & SignalWatcherApi;
+}
+
+let effectsPending = false;
+const effectWatcher = new Signal.subtle.Watcher(() => {
+  if (effectsPending) {
+    return;
+  }
+  effectsPending = true;
+  queueMicrotask(() => {
+    effectsPending = false;
+    for (const signal of effectWatcher.getPending()) {
+      signal.get();
+    }
+    effectWatcher.watch();
+  });
+});
+
+/**
+ * Executes the provided callback function when any of the signals it accesses
+ * change. If an options object is provided, the `element` property can be used
+ * to specify the element to associate the effect with. The `beforeUpdate`
+ * property can be used to specify that the effect should run before the element
+ * updates. The `manualDispose` property can be used to specify that the effect
```

Comment:
What's the use case for `manualDispose`?

Reply:
Clarified docs

Reply:
Added test for `manualDispose`